packages/web-fetch/src/index.ts

```diff
@@ -0,0 +1,96 @@
+import {reatomAsync} from '@reatom/async'
+import {Atom, Ctx, isAtom, throwReatomError} from '@reatom/core'
+
+export interface ReatomFetchOptions<O, I> {
+	fetch?: (url: string, init: RequestInit) => Promise<Response>
+	init?: Omit<RequestInit, 'method' | 'signal' | 'body' | 'headers'>
+	auth?: MaybeAtom<string>
+	headers?: MaybeAtom<HeadersInit>
+	baseUrl?: string
+	method?: string
+	validate?: (o: O) => boolean | string
+	parseResponse?: (response: Response) => Promise<O>
+	serializeBody?: (i: I, headers: Headers) => BodyInit
+	validateStatus?: (status: number) => boolean | string
+}
+
+type UrlSearchParamsInit = ConstructorParameters<typeof URLSearchParams>[0]
+
+type MaybeAtom<T> = Atom<T> | T
+
+const defaults: ReatomFetchOptions<unknown, unknown> = {
+	init: {},
+	fetch,
+	validate: () => true,
+	validateStatus: status => status >= 200 && status < 300,
+	parseResponse: response => {
+		const ct = response.headers.get(`content-type`)
+		if (ct !== `application/json`) {
+			throw new Error(`Invalid response content-type "${ct}"`)
+		}
+		return response.json()
+	},
+	serializeBody: body => JSON.stringify(body),
+}
+
+export const createReatomFetch = (
+	options: ReatomFetchOptions<unknown, unknown>
+) => {
+	const baseOpts = {...defaults, ...options}
+
+	return <O, I = void>(url: string, opts: ReatomFetchOptions<O, I> = {}) =>
+		reatomAsync(async (ctx, i: I, params?: UrlSearchParamsInit) => {
```

Comment:
What `i` mean, "input"?
```suggestion
		reatomAsync(async (ctx, params: I, search?: UrlSearchParamsInit) => {
```